cmd/loom/loom.go

```diff
@@ -1133,9 +1136,11 @@ func deployContract(
 		return err
 	}
 
-	err = registry.Register(contractCfg.Name, addr, addr)
-	if err != nil {
-		return err
+	if chainCfg.EvmConfig.AllowNamedEvmContract || vmType == lvm.VMType_PLUGIN {
```

Comment:
This `deployContract` function doesn't need to be changed at all, it only gets called when a new chain is initialized to deploy contracts specified in the chain genesis, and we can assume whoever writes the genesis knows what they're doing.